### PR TITLE
Version history

### PR DESCRIPTION
**Pull request by SjoenH**
> _**User-story:**_ As an end-user I want to keep track of all changes and possibly being able to roll-back to earlier versions to be able to understand history and secure understanding for where we are today and why (continuous improvement).

You may now look at previous versions of a process and see what changed over time.

closes https://github.com/equinor/MAD-VSM-WEB/issues/154

---
### Requirements
Unless mentioned otherwise, any user can do the following:

**Current**
- [x]     User can see the current version.
- [x]     User (with edit-role) can edit a card in the current version.
- [x]     User can see the details of any card in the current version.

**Previous**
- [x]  User can see a list of all versions by clicking on the version icon (down right-hand corner). (versions are ordered by date and only one per day is shown)
- [x] User can switch between showing the changes grouped by Year, Month, Day, Hours, Minutes or Seconds (default to Days)
- [x]     User can view a previous version.
- [x]     User can view the details of any card in a previous version.
- [x]     User can share the url of any version with others (copy from the url bar) and opening the url in a new tab will show that version, with the version view visible.
- [x]     User cannot edit a card in a previous version. ( Drag'n drop interactions should be disabled, and sidebar view should be in read-only mode. )
- [x]     Exiting the version view should return the user to the current version.

### CODE REVIEW

**Comment by SjoenH** [comment]
@dansvend Kinda big PR. Let me know if something is not clear 😅

**Review by dansvend — APPROVED** [review]
Looks good🔥

File: components/CategorizationPageButton.tsx [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
+
+export const VersionHistoryButton = (props: {
+  handleVersionHistoryClick: () => void;
+}): JSX.Element => (
+  <ButtonWrapper
+    icon={timeline}
+    title={"Show version history"}
+    onClick={props.handleVersionHistoryClick}
+  />
+);
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/CategorizationPageButton.tsx]: Done

File: utils/GetDateTimeOption.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+//Get DateTime Option for a given resolution
+export function getDateTimeOption(
+  resolution: "year" | "month" | "day" | "hour" | "minute" | "second" | string
+): Intl.DateTimeFormatOptions {
+  if (resolution === "year") {
+    return { year: "2-digit" };
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply utils/GetDateTimeOption.ts]: Done

File: components/CanvasButtons.tsx [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
   handleClickLabel: () => void;
+  handleClickVersionHistory: () => void;
 }): JSX.Element => {
   return (
     <div className={styles.wrapper}>
       {props.userCanEdit && (
         <ManageLabelButton handleClickLabel={props.handleClickLabel} />
       )}
-      <CategorizationPageButton />
+      <CategorizationPageButton userCanEdit={props.userCanEdit} />
```
[the remark this answers is not in the document]
Comment (SjoenH) [reply components/CanvasButtons.tsx]: Nope, fixed

**Comment by SjoenH** [comment]
Waiting on testing before squash and merge

**Comment by SjoenH** [comment]
Released to Dev, Test & QA

**Comment by SjoenH** [comment]
Fixed a bug where we always fetched the new version of a selected vsmObject/card. Even though it might be different or not exist at all. @dansvend please have a look over the new changes 👍

**Review by dansvend — APPROVED** [review]
Looks good💪

**Pull request merged**